src/server/middleware/repoproxy/proxy.go

```diff
@@ -33,18 +34,21 @@ import (
 	httpLib "github.com/goharbor/harbor/src/lib/http"
 	"github.com/goharbor/harbor/src/lib/log"
 	"github.com/goharbor/harbor/src/lib/orm"
+	"github.com/goharbor/harbor/src/lib/redis"
 	proModels "github.com/goharbor/harbor/src/pkg/project/models"
+	"github.com/goharbor/harbor/src/pkg/proxy/connection"
 	"github.com/goharbor/harbor/src/pkg/reg/model"
 	"github.com/goharbor/harbor/src/server/middleware"
 )
 
 const (
-	contentLength       = "Content-Length"
-	contentType         = "Content-Type"
-	dockerContentDigest = "Docker-Content-Digest"
-	etag                = "Etag"
-	ensureTagInterval   = 10 * time.Second
-	ensureTagMaxRetry   = 60
+	contentLength              = "Content-Length"
+	contentType                = "Content-Type"
+	dockerContentDigest        = "Docker-Content-Digest"
+	etag                       = "Etag"
+	ensureTagInterval          = 10 * time.Second
+	ensureTagMaxRetry          = 60
+	upstreamRegistryLimitLevel = "UPSTREAM_REGISTRY_LIMIT_LEVEL" // if UPSTREAM_REGISTRY_LIMIT_LEVEL is project, the upstream registry connection is based on project level, by default it is artifact level
```

Comment:
I have a general question, 

1. When the environment is set to project, does this mean that all /v2 requests within this project will be counted?
2. When the environment is set to artifact, does this mean that all /v2 requests for this artifact will be counted?

So, I think we need a concise describe on this env of the meaning.

Reply:
Yes. 
1. all GET/HEAD request to manifest/blobs to the same project is counted.
2. all GET/HEAD request to manifest/blobs to the same artifact is counted.
2.